File: Builder/PHP/XmlBuilder.php
<?php
  class XmlBuilder implements Builder {
    private $content;

    public function buildHeader($header) {
      $this->content .= "<Header>" . $header . "</Header>\n";
    }

    public function buildBody($body) {
      $this->content .= "<Body>" . $body . "</Body>\n";
    }

    public function buildFooter($footer) {
      $this->content .= "<Footer>" . $footer . "</Footer>\n";
    }

    public function getResult() {
      return $this->content;
    }
  }

File: Builder/Go/builder/XmlBuilder.go
package builder

type XmlBuilder struct {
  content string
}

func (this *XmlBuilder) buildHeader(header string) {
  this.content += "<Header>" + header + "</Header>\n"
}

func (this *XmlBuilder) buildBody(body string) {
  this.content += "<Body>" + body + "</Body>\n"
}

func (this *XmlBuilder) buildFooter(footer string) {
  this.content += "<Footer>" + footer + "</Footer>\n"
}

func (this *XmlBuilder) GetResult() string {
  return this.content
}
